packages/react-components/react-combobox/src/components/Combobox/useComboboxStyles.ts

```diff
@@ -1,30 +1,167 @@
+    ':focus-within': {
+      outlineWidth: '2px',
+      outlineStyle: 'solid',
+      outlineColor: 'transparent',
+    },
+
+    // bottom focus border, shared with Input, Select, and SpinButton
+    '::after': {
+      boxSizing: 'border-box',
+      content: '""',
+      position: 'absolute',
+      left: '-1px',
+      bottom: '-1px',
+      right: '-1px',
+      height: `max(2px, ${tokens.borderRadiusMedium})`,
+      borderBottomLeftRadius: tokens.borderRadiusMedium,
+      borderBottomRightRadius: tokens.borderRadiusMedium,
+      ...shorthands.borderBottom('2px', 'solid', tokens.colorCompoundBrandStroke),
+      clipPath: 'inset(calc(100% - 2px) 0 0 0)',
+      transform: 'scaleX(0)',
+      transitionProperty: 'transform',
+      transitionDuration: tokens.durationUltraFast,
+      transitionDelay: tokens.curveAccelerateMid,
+    },
+    ':focus-within::after': {
+      transform: 'scaleX(1)',
+      transitionProperty: 'transform',
+      transitionDuration: tokens.durationNormal,
+      transitionDelay: tokens.curveDecelerateMid,
+    },
+    ':focus-within:active::after': {
+      borderBottomColor: tokens.colorCompoundBrandStrokePressed,
+    },
+  },
+
   listbox: {},
-  trigger: {},
+
+  input: {
+    backgroundColor: tokens.colorTransparentBackground,
+    ...shorthands.border('0'),
+    width: '100%',
+
+    '&:focus': {
+      outlineStyle: 'none',
+    },
+  },
+
+  // size variants
+  small: {
+    fontSize: tokens.fontSizeBase200,
+    lineHeight: tokens.lineHeightBase200,
+    ...shorthands.padding('3px', tokens.spacingHorizontalSNudge),
+  },
+  medium: {
+    fontSize: tokens.fontSizeBase300,
+    lineHeight: tokens.lineHeightBase300,
+    ...shorthands.padding('5px', tokens.spacingHorizontalMNudge),
+  },
+  large: {
+    columnGap: tokens.spacingHorizontalSNudge,
+    fontSize: tokens.fontSizeBase400,
+    lineHeight: tokens.lineHeightBase400,
+    ...shorthands.padding('7px', tokens.spacingHorizontalM),
+  },
+
+  // appearance variants
+  outline: {
+    backgroundColor: tokens.colorNeutralBackground1,
+    ...shorthands.border('1px', 'solid', tokens.colorNeutralStroke1),
+    borderBottomColor: tokens.colorNeutralStrokeAccessible,
+  },
+  underline: {
+    backgroundColor: tokens.colorTransparentBackground,
+    ...shorthands.borderBottom('1px', 'solid', tokens.colorNeutralStrokeAccessible),
+    ...shorthands.borderRadius(0),
+  },
+  'filled-lighter': {
+    backgroundColor: tokens.colorNeutralBackground1,
+  },
+  'filled-darker': {
+    backgroundColor: tokens.colorNeutralBackground3,
+  },
+});
+
+const useIconStyles = makeStyles({
+  icon: {
+    boxSizing: 'border-box',
+    color: tokens.colorNeutralStrokeAccessible,
+    display: 'block',
+    flexGrow: 0,
+    flexShrink: 0,
+    fontSize: tokens.fontSizeBase500,
+
+    // the SVG must have display: block for accurate positioning
+    // otherwise an extra inline space is inserted after the svg element
+    '& svg': {
+      display: 'block',
+    },
```

Comment:
Is this a problem that could be common to all of our icon slots? Or is it something odd for just Combobox?

Reply:
I'd expect it to be common to our icon slots that don't want an svg to be inline with text, and where extra space characters would matter.

Reply:
unless something has changed with the icons -- I wrote this a while ago, and just copied it from ComboButton in this PR :D